vignettes/sim_fixed_design_custom.Rmd

```diff
@@ -0,0 +1,406 @@
+
+The following code lists all possible tests available in `simtrial`. Users can select one of the tests listed above or combine the testing results to make comparisons across tests. For demonstration purposes, we will aggregate all tests together.
+```{r}
+# Logrank test
+sim_res_lr <- cut_data |> wlr(weight = fh(rho = 0, gamma = 0))
+
+# weighted logrank test by Fleming-Harrington weights
+sim_res_fh <- cut_data |> wlr(weight = fh(rho = 0, gamma = 0.5))
+
+# Modestly weighted logrank test
+sim_res_mb <- cut_data |> wlr(weight = mb(delay = Inf, w_max = 2))
+
+# Weighted logrank test by Xu 2017's early zero weights
+sim_res_xu <- cut_data |> wlr(weight = early_zero(early_period = 3))
+
+# RMST test
+sim_res_rmst <- cut_data |> rmst(tau = 10)
+
+# Milestone test
+sim_res_ms <- cut_data |> milestone(ms_time = 10)
+
+# Maxcombo tests comboing multiple weighted logrank test with Fleming-Harrington weights
+sim_res_mc <- cut_data |> maxcombo(rho = c(0, 0), gamma = c(0, 0.5))
+```
+
+The output of the tests mentioned above are lists including:
+
+- The testing method employed (WLR, RMST, milestone, or MaxCombo), which can be accessed using `sim_res_rmst$method`.
+- The parameters associated with the testing method. For instance, the RMST test parameter is 10, indicating that the RMST is evaluated at month 10. You can find this information using `sim_res_rmst$parameter`.
+- The point estimate and standard error for the testing method used. For example, the point estimate for RMST represents the survival difference between the experimental group and the control group. This estimate can be retrieved with `sim_res_rmst$estimate` and `sim_res_rmst$se`.
+- The Z-score for the testing method, accessible via `sim_res_rmst$z`. Please note that the Z-score is not provided for the MaxCombo test; instead, the p-value is reported (`sim_res_mc$p_value`).
+
+```{r}
+sim_res <- tribble(
+  ~Method, ~Parameter, ~Z, ~Estimate, ~SE, ~`P value`,
+  sim_res_lr$method, sim_res_lr$parameter, sim_res_lr$z, sim_res_lr$estimate, sim_res_lr$se, pnorm(-sim_res_lr$z),
+  sim_res_fh$method, sim_res_fh$parameter, sim_res_fh$z, sim_res_fh$estimate, sim_res_fh$se, pnorm(-sim_res_fh$z),
+  sim_res_mb$method, sim_res_mb$parameter, sim_res_mb$z, sim_res_mb$estimate, sim_res_mb$se, pnorm(-sim_res_mb$z),
+  sim_res_xu$method, sim_res_xu$parameter, sim_res_xu$z, sim_res_xu$estimate, sim_res_xu$se, pnorm(-sim_res_xu$z),
+  sim_res_rmst$method, sim_res_rmst$parameter|> as.character(), sim_res_rmst$z, sim_res_rmst$estimate, sim_res_rmst$se, pnorm(-sim_res_rmst$z),
+  sim_res_ms$method, sim_res_ms$parameter |> as.character(), sim_res_ms$z, sim_res_ms$estimate, sim_res_ms$se, pnorm(-sim_res_ms$z),
+  sim_res_mc$method, sim_res_mc$parameter, NA, NA, NA, sim_res_mc$p_value
+  ) 
```

Comment:
I see here is where you directly call `tribble()` from the {tibble} package. To keep this usage, please add {tibble} to the field Suggests in `DESCRIPTION`

Reply:
Thanks for checking. I added it back.